packages/astro/src/@types/astro.ts

```diff
@@ -2553,6 +2553,7 @@ export interface RouteData {
 	segments: RoutePart[][];
 	type: RouteType;
 	prerender: boolean;
+	hasSharedModules: boolean;
```

Comment:
I don't think I'd use routeData for this, given that it's not a property of the route itself, it's more like internal state for the build. Normally we do this sort of thing either using the build internals or the PageData (I'd probably go with PagaData here).